server/global/data_analyzer/analyzer.js

```diff
@@ -0,0 +1,298 @@
+import levenshtein from "js-levenshtein";
+
+import { MAX_LENGTH } from "../data_importer/dataImporter.js";
+
+const DCI_DISTANCE_MIN = 1;
+const PROPERTY_VALUE_MIN_DISTANCE = 2;
+const CLASSIFICATION_VALUE_MIN_DISTANCE = 2;
+
+/**
+ * Analyse the data object and returns warnings
+ * @param {object} data
+ * @returns {AnalyzerWarning[]}
+ */
+export function analyzeData(data) {
+  const warnings = [];
+
+  warnings.push(...analyzeClassification("classes", data.classes));
+  warnings.push(...analyzeClassification("systèmes", data.systems));
+
+  warnings.push(...analyzeProperty("indications", data.indications, 128));
+  warnings.push(...analyzeProperty("effets indésirables", data.side_effects, 128));
+  warnings.push(...analyzeProperty("intéractions", data.interactions, 128));
+
+  warnings.push(...analyzeMolecules(data.molecules));
+
+  return warnings;
+}
+
+/**
+ * Analyze a property
+ * @param {string} property The property name
+ * @param {{id : number, name : string}[]} values
+ * @returns {AnalyzerWarning[]}
+ */
+function analyzeProperty(property, values) {
+  const names = values.map((v) => v.name);
+  const tooLongValues = getTooLongValues(names, MAX_LENGTH.PROPERTY_VALUE);
+  const closeValues = getTooCloseValues(names, PROPERTY_VALUE_MIN_DISTANCE);
+  const noStringValue = names.filter((name) => !isString(name));
+
+  return [
+    ...tooLongValues.map(
+      (value) =>
+        new AnalyzerWarning(
+          AnalyzerWarning.TOO_LONG_VALUE,
+          `La valeur '${value}' de la propriété '${property}' est trop long (max ${MAX_LENGTH.PROPERTY_VALUE})`
```

Comment:
If this message will be displayed to the user, perhaps it would be better to put double quotes (instead of single ones)